src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/MessagePackHubProtocol.cs

```diff
@@ -81,22 +84,43 @@ private static InvocationMessage CreateInvocationMessage(Unpacker unpacker, IInv
             var invocationId = ReadInvocationId(unpacker);
             var nonBlocking = ReadBoolean(unpacker, "nonBlocking");
             var target = ReadString(unpacker, "target");
-            var argumentCount = ReadArrayLength(unpacker, "arguments");
             var parameterTypes = binder.GetParameterTypes(target);
 
+            try
+            {
+                var arguments = BindArguments(unpacker, parameterTypes);
+                return new InvocationMessage(invocationId, nonBlocking, target, argumentBindingException: null, arguments: arguments);
+            }
+            catch (Exception ex)
+            {
+                return new InvocationMessage(invocationId, nonBlocking, target, ExceptionDispatchInfo.Capture(ex));
+            }
+        }
+
+        private static object[] BindArguments(Unpacker unpacker, Type[] parameterTypes)
+        {
+            var argumentCount = ReadArrayLength(unpacker, "arguments");
+
             if (parameterTypes.Length != argumentCount)
             {
                 throw new FormatException(
-                    $"Target method expects {parameterTypes.Length} arguments(s) but invocation has {argumentCount} argument(s).");
+                    $"Invocation provides {argumentCount} argument(s) but target expects {parameterTypes.Length}.");
```

Comment:
Do we log on the server for any binding failures?

Reply:
Yes - it now fails when we are invoking the method so it is treated as any other hub method invocation error